shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/SurveyDetailViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package co.nimblehq.avishek.phong.kmmic.presentation.module
+
+import co.nimblehq.avishek.phong.kmmic.data.remote.helper.extension.toErrorMessage
+import co.nimblehq.avishek.phong.kmmic.domain.model.Survey
+import co.nimblehq.avishek.phong.kmmic.domain.usecase.GetSurveyDetailUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.update
+
+data class SurveyDetailViewState(
+    val isLoading: Boolean = false,
+    val errorMessage: String? = null,
+    val survey: Survey? = null
+
+) {
+    constructor() : this(false)
```

Comment:
why we need this? can we do this as isLoading is false as default:
```suggestion
    constructor() : this()
```

Reply:
Solved in ecb0b79